include/pipe.h

```diff
@@ -0,0 +1,96 @@
+#ifndef PIPE_NETWORK_PIPE_H_
+#define PIPE_NETWORK_PIPE_H_
+
+#include <cmath>
+
+#include <array>
+#include <exception>
+#include <memory>
+
+#include "node.h"
+#include "settings.h"
+
+namespace pipenetwork {
+class Pipe;
+}
+
+//! Pipe class
+//! \brief Class that stores the information about pipes
+class pipenetwork::Pipe {
+
+ public:
+  // Constructor with id and node pointers
+  //! \param[in] id node id
+  //! \param[in] nodes array of node pointers
+  Pipe(unsigned id,
+       const std::array<std::shared_ptr<pipenetwork::Node>, 2>& nodes);
+
+  //! Destructor
+  ~Pipe() { nodes_.fill(nullptr); }
+
+  //! Copy constructor
+  Pipe(const Pipe&) = delete;
+
+  //! Assignment operator
+  Pipe& operator=(const Pipe&) = delete;
+
+  //! Move constructor
+  Pipe(Pipe&&) = delete;
+
+  //! Return id
+  //! \retval id_ id of the pipe
+  unsigned id() const { return id_; }
+
+  //! Assign radius of the pipe
+  //! \param[in] radius radius of the pipe
+  void radius(double radius) { radius_ = radius; }
+
+  //! Return length of the pipe
+  //! \retval length_ pipe length
+  double length() { return length_; }
+
+  //! Assign Dracy friction factor
+  //! \param[in] darcy_friction Darcy friction factor of the pipe
+  void darcy_friction(double darcy_friction) {
+    darcy_friction_ = darcy_friction;
+  }
+
+  //! Calculate and return discharge using Darcy-Weisbach equation:
+  //! dhead = (8*darcy_factor*pow(discharge,2)/(pow(M_PI,2)*g*pow(2*radius,5))
+  //! That is, discharge =
+  //! sqrt(dhead*pow(M_PI,2)*g*pow(2*radius,5)/(8*darcy_friction)); \retval
+  //! discharge_ discharge in the pipe
```

Comment:
`//! \retval`

Reply:
Add what the function does

Reply:
What the function does was stated at the first line of the part of comment